src/main/python/setup.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# -------------------------------------------------------------
+#
+# Modifications Copyright 2020 Graz University of Technology
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# -------------------------------------------------------------
+
+from __future__ import print_function
+import os
+import sys
+from setuptools import find_packages, setup
+import time
+
+try:
+    exec(open('systemds/project_info.py').read())
+except IOError:
+    print("Could not read project_info.py. Will use default values.", file=sys.stderr)
+    BUILD_DATE_TIME = str(time.strftime("%Y%m%d.%H%M%S"))
+    __project_artifact_id__ = 'systemds'
+    __project_version__ = BUILD_DATE_TIME + '.dev0'
+ARTIFACT_NAME = __project_artifact_id__
+ARTIFACT_VERSION = __project_version__
+ARTIFACT_VERSION_SHORT = ARTIFACT_VERSION.split("-")[0]
+
+numpy_version = '1.8.2'
+py4j_version = '0.10.0'
```

Comment:
You might want to make a requirements.txt also, for easy standard installation.

https://stackoverflow.com/questions/7225900/how-to-install-packages-using-pip-according-to-the-requirements-txt-file-from-a#15593865

Reply:
Setuputils creates the requirements file automatically, that`s what the `setup.py` is for.